File: iteration1/Tests/PersonTest.java
package iteration1.Tests;

import iteration1.Advisor;
import iteration1.Person;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class PersonTest {

    private Person person;

    @Before
    public void setUp() {
        person = new Advisor(1, "name", "surname", "validUsername", "validPassword");
    }

    @Test
    public void loginShouldReturnTrueWhenCredentialsAreValid() {
        String username = "validUsername";
        String password = "validPassword";
        assertTrue(person.login(username, password));
    }

    @Test
    public void loginShouldReturnFalseWhenCredentialsAreInvalid() {
        String username = "invalidUsername";
        String password = "invalidPassword";
        assertFalse(person.login(username, password));
    }
}

File: iteration3/Test/PersonTest.py
import unittest

from iteration3.Advisor import Advisor


class PersonTest(unittest.TestCase):

    def setUp(self):
        self.person = Advisor(1, "name", "surname", "validUsername", "validPassword")

    def test_loginShouldReturnTrueWhenCredentialsAreValid(self):
        username = "validUsername"
        password = "validPassword"
        self.assertTrue(self.person.login(username, password))

    def test_loginShouldReturnFalseWhenCredentialsAreInvalid(self):
        username = "invalidUsername"
        password = "invalidPassword"
        self.assertFalse(self.person.login(username, password))


if __name__ == "__main__":
    unittest.main()
